### PR TITLE
feat: add beacon api headers types

### PR DESCRIPTION
**Pull request by mattsse**
add missing types for:

https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockHeader

### CODE REVIEW

File: crates/rpc/rpc-types/src/beacon/header.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    pub finalized: bool,
+    /// Container for the header data.
+    pub data: HeaderData,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct HeaderData {
+    /// root hash
+    pub root: B256,
+    /// whether the block is canonical
```
Comment (onbjerg) [inline crates/rpc/rpc-types/src/beacon/header.rs:41]:
```suggestion
    /// Whether the block is part of the canonical chain
```

File: crates/rpc/rpc-types/src/beacon/header.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    /// choice. If the field is not present, additional calls are necessary to compare the epoch of
+    /// the requested information with the finalized checkpoint.
+    pub finalized: bool,
+    /// Container for the header data.
+    pub data: HeaderData,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct HeaderData {
+    /// root hash
```
Comment (onbjerg) [inline crates/rpc/rpc-types/src/beacon/header.rs:39]: root of what?

**Review by onbjerg — APPROVED** [review]
smol doc nits

**Review by Evalir — APPROVED** [review]
Should this be upstreamed to alloy?

**Comment by mattsse** [comment]
> Should this be upstreamed to alloy?

eventually as beacon types or something

**Pull request merged**